velox/exec/Unnest.cpp

```diff
@@ -192,13 +192,25 @@ RowVectorPtr Unnest::getOutput() {
       // Construct unnest column using Array elements wrapped using above
       // created dictionary.
       auto unnestBaseArray = currentDecoded.base()->as<ArrayVector>();
-      outputs[outputsIndex++] = identityMapping
-          ? unnestBaseArray->elements()
-          : wrapChild(
-                numElements,
-                elementIndices,
-                unnestBaseArray->elements(),
-                nulls);
+      auto elements = unnestBaseArray->elements();
+      auto type = elements->type();
+      if (type->isRow()) {
+        auto size = type->size();
+        auto row = elements->as<RowVector>();
```

Comment:
It seems like this is still going to crash if elements of the array are encoded as dictionary or const. Would you add a test with such input?

Reply:
The existing codepath handles the dictionary and constant encodings. Added tests with array of maps and constant arrays, thanks.

Reply:
The existing code doesn't assume that elements is an instance of RowVector. It can be a DictionaryVector or a ConstantVector. In this case elements->as<RowVector>() will fail. Please, take a closer look.

> Added tests with array of maps and constant arrays, thanks.

Make sure to test with array vectors whose elements vector is a dictionary or a constant.

Reply:
Thanks for the feedback @mbasmanova, and apologies for the delay in addressing this. Tests for `elements` having constant and map encodings are added. I believe `currentDecoded.base()` ensures the elements of `unnestBaseArray` do not have dictionary encoding; I was not able to get dictionary encoding for `elements` by wrapping the array elements in a dictionary, similar to the existing test `arrayWithOrdinality`. Please let me know if I am missing something here.

Reply:
@pramodsatya Thank you for looking into this. Can you point me to the test that create an array with dictionary-encoded elements?

Reply:
The tests `arrayWithOrdinality`, `mapWithOrdinality`, and `complexArrays` have arrays with dictionary encoded elements. The VectorPtr `elements` however does not have dictionary encoding for any of the tests in `UnnestTest.cpp`.

Reply:
Apologies for the confusion, I was referring to the tests with dictionary encoded array. I am able to repro the error you were referring to and will add the fix + tests for this case.

Reply:
@pramodsatya Happy to hear you are able to trigger this use case.

Reply:
Thanks for pointing out this case @mbasmanova, it is addressed now. A check is added in `Unnest` to ensure `rowVector` is not null along with tests for dictionary encoded elements. 
Could you please help review the changes?